src/Azure/Orleans.ScheduledJobs.AzureStorage/NetstringJsonSerializer.cs

```diff
@@ -0,0 +1,163 @@
+using System;
+using System.Buffers;
+using System.Buffers.Text;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization.Metadata;
+using System.Threading.Tasks;
+
+namespace Orleans.ScheduledJobs.AzureStorage;
+
+/// <summary>
+/// Provides methods for serializing and deserializing JSON data using the netstring format.
+/// Netstrings are a simple, self-delimiting way to encode data with length prefixes.
+/// Format: [length]:[data]\n
+/// </summary>
+public static class NetstringJsonSerializer
+{
+    /// <summary>
+    /// Encodes an object as a netstring by serializing it to JSON.
+    /// </summary>
+    /// <typeparam name="T">The type of the object to encode.</typeparam>
+    /// <param name="value">The object to encode.</param>
+    /// <param name="jsonTypeInfo">The JSON type info for serialization.</param>
+    /// <returns>A byte array containing the netstring-encoded JSON data.</returns>
+    public static byte[] Encode<T>(T value, JsonTypeInfo<T> jsonTypeInfo)
+    {
+        // Serialize to JSON using ArrayBufferWriter to avoid allocations
+        var bufferWriter = new ArrayBufferWriter<byte>();
+        using (var writer = new Utf8JsonWriter(bufferWriter))
+        {
+            JsonSerializer.Serialize(writer, value, jsonTypeInfo);
+        }
+
+        var jsonBytes = bufferWriter.WrittenSpan;
+        var jsonLength = jsonBytes.Length;
+
+        // Calculate length prefix size
+        Span<byte> lengthBuffer = stackalloc byte[20]; // Max int32 digits + colon
```

Comment:
If we use hex & pad it out to the max length, then we can avoid the copying & allocations here. We can take a stream as input and seek back to the initial position to write the length.

Reply:
We could use `PooledBufferStream` to write to